codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/AddAwsConfigFields.java

```diff
@@ -134,42 +151,46 @@ public void writeAdditionalFiles(
 
     @Override
     public List<RuntimeClientPlugin> getClientPlugins() {
-        return ListUtils.of(
-                RuntimeClientPlugin.builder()
-                        .configFields(UNIVERSAL_FIELDS)
-                        .build(),
-                RuntimeClientPlugin.builder()
-                        .configFields(ListUtils.of(ConfigField.builder()
-                                .name(CREDENTIALS_CONFIG_NAME)
-                                .type(getAwsCoreSymbol("CredentialsProvider"))
-                                .documentation("The credentials object to use when signing requests.")
-                                .build()))
-                        .servicePredicate((model, serviceShape) -> model.getKnowledge(ServiceIndex.class)
-                                .getAuthSchemes(serviceShape).values().stream().anyMatch(trait -> trait.getClass()
-                                        .equals(SigV4Trait.class)))
-                        .build()
-        );
+        List<RuntimeClientPlugin> plugins = new ArrayList<>();
+
+        List<ConfigField> allClients = AWS_CONFIG_FIELDS.stream().filter(AwsConfigField::isGeneratedOnClient)
+                .filter(field -> !field.getServicePredicate().isPresent())
```

Comment:
Why is this filter being applied?

Reply:
This is strictly to group all config fields that don't have a specified service predicate into a single RuntimeClientPlugin. Versus generating a RuntimeClientPlugin per config field.

Reply:
That's fine, just add a comment and it's good